### ISSUE TITLE
HW issues

### ISSUE THREAD

**Issue by Malar-Natarajan** [openstax/tutor#326]
### Description

1. The HW step numbers are starting from 0 instead of 1. The breadcrumbs start from 1 so there is a mismatch in numbers between breadcrumbs and question numbers

2. For assignments with immediate feedback the breadcrumbs are not updating the feedback though the exercise updates immediately. But as we proceed few questions at some point all breadcrumbs get updated. See screenshot
https://drive.google.com/file/d/1d_Ym4lbgFPN_c5j6uM3g1q_owZKX8vlf/view?usp=sharing

3. The question number for a HW displays only after the FR is entered whereas in staging the question has its number as soon as the question is loaded
https://drive.google.com/file/d/1s_WmnIThsfjDOW7ca3yQysm0NjLVdxyF/view?usp=sharing

4. Received a 422 error "unknown step type placeholder" while doing a HW with feedback after due setup

5. placeholder for the individual review card is missing in the breadcrumbs for a HW with "feedback after due" setup

![Screen Shot 2019-04-04 at 3.39.40 PM.png](https://images.zenhubusercontent.com/5b23e2cc835fb47f522b81ba/421be4b5-e6ca-4866-95b9-587e8219724a)

![Screen Shot 2019-04-04 at 3.41.40 PM.png](https://images.zenhubusercontent.com/5b23e2cc835fb47f522b81ba/e86d0601-e456-4716-a372-fcaa08ea019d)

**To Reproduce**
Steps to reproduce the behavior are

**Expected behavior**

**Screenshots**

**Additional context**

### Acceptance Tests
Additional acceptance tests that are required by this bug fix.

**title**: template  
**categories**: Interactive Component, Navigation, TOC, Math, Book Content, Other
> GIVEN something  
> AND something else  
> WHEN something  
> AND something else  
> THEN something  
> AND something else  

### Checklist for Done
2-CODE
  - [ ] A pull request is opened and linked to this issue. The automated pull request checks pass. 
  - [ ] The change has been approved by other developers.
  - [ ] The pull request is merged into master, and the change branch is deleted.
  - [ ] If there are new acceptance tests, they are finalized in "Given / When / Then" format.
  - [ ] Regression test categories are identified on the issue. Categories should be referenced if any changes might affect them, even if the intended functionality is unchanged.
- [ ] Add milestone tag
  
4A-UX REVIEW
- [ ] If UI changes, UX reviews and verifies. 

5A-FUNCT VER
  - [ ] All the acceptance tests for this issue have passed.
 
5C-REGRESSION
  - [ ] The test plan in testrail passes

**Comment by nathanstitt** [comment]
These were fixed on one of the other task step fixes cards.  please re-open if you notice anything not working.

**Comment by Brooke-Fail** [comment]
Verified in QA.

**Comment by Brooke-Fail** [comment]
Verified in Staging.

**Comment by Brooke-Fail** [comment]
Verified in Production.
